### PR TITLE
feat(hermes): move to wormhole gRPC from P2P

### PR DESCRIPTION
**Pull request by Reisen**
This PR removes the current p2p code used to listen to Wormhole's P2P network and replaces it with the Wormhole gRPC definitions, using a Wormhole node gRPC endpoint to listen for VAAs instead. This let's us use standard wormhole nodes as our source of data and also allows us to move to more robust/load balanced solutions.

- Removes P2P options from `Config` and adds `gRPC` address.
- Removes old `wormhole.rs` module in favor of `network/wormhole.rs`
- Adds gRPC definitions from upstream Wormhole for using gRPC endpoints.
- Replaces p2p service with wormhole service for fetching and processing VAAs.

### CODE REVIEW

File: hermes/src/config/wormhole.rs [hunk 1]
```diff
@@ -1,48 +1,27 @@
...
-    pub listen_addrs: Vec<Multiaddr>,
-
-    /// Network ID for Wormhole
-    #[arg(long = "wormhole-network-id")]
-    #[arg(default_value = DEFAULT_NETWORK_ID)]
-    #[arg(env = "WORMHOLE_NETWORK_ID")]
-    pub network_id: String,
+    /// gRPC endpoint for a Wormhole node.
+    ///
+    /// This can either be a standard Wormhole node gRPC endpoint or a beacon endpoint if
```
Comment (ali-bahjati) [inline hermes/src/config/wormhole.rs:21]: does node has the same api as `spy`? maybe call it spy_rpc_addr :?

Comment (thmzlt) [reply hermes/src/config/wormhole.rs:21]: I agree.

File: hermes/src/network/wormhole.rs [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+            tracing::error!(error = ?e, "Wormhole gRPC service failed.");
+        }
+
+        tokio::time::sleep(tokio::time::Duration::from_secs(5)).await;
+    }
+    Ok(())
+}
+
+async fn run(state: Arc<State>) -> Result<()> {
+    let mut client = SpyRpcServiceClient::connect("grpc://127.0.0.1:7073").await?;
```
Comment (thmzlt) [inline hermes/src/network/wormhole.rs:159]: Maybe pass the address as a value from the outside?

File: hermes/src/config/wormhole.rs (hunk 1 above)
Comment (Reisen) [reply hermes/src/config/wormhole.rs:21]: Hmm, I thought the goal was to have them have the same API?

File: hermes/src/network/wormhole.rs (hunk 2 above)
Comment (Reisen) [reply hermes/src/network/wormhole.rs:159]: That was the intent I missed this, will fix.

File: hermes/src/network/wormhole.rs [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+        vaa.emitter_chain == Chain::Pythnet,
+        "VAA from non-Pythnet Chain."
+    );
+    ensure!(
+        vaa.emitter_address == Address(ACCUMULATOR_EMITTER_ADDRESS),
+        "VAA from non-Pythnet Emitter: {}",
+        vaa.emitter_address
+    );
+    Ok(())
+}
```
Comment (ali-bahjati) [inline hermes/src/network/wormhole.rs:231]: we might be able to move this to filters for spy

**Review by ali-bahjati — APPROVED** [review]

**Pull request merged**